src/service/xhr-impl.js

```diff
@@ -265,6 +267,23 @@ export class Xhr {
   getCorsUrl(win, url) {
     return getCorsUrl(win, url);
   }
+
+  /**
+   * Sets up and normalizes the FetchInitDef
+   *
+   * @param {?FetchInitDef=} opt_init Fetch options object.
+   * @param {string=} opt_accept The HTTP Accept header value.
+   * @return {!FetchInitDef}
+   */
+  setupInit_(opt_init, opt_accept) {
```

Comment:
With https://github.com/ampproject/amphtml/pull/7562/commits/d61c81f1d82a92128826bfcd9f42260c1a7d200e#r103352978, it won't be necessary. You'll just call `Xhr#fetchJson`, and cache the result if cacheable.